### PR TITLE
SNOW-853533: Support case sensitivity in to_local_iterator

### LINKED ISSUE #926: SNOW-853533: to_local_iterator doesn't support generating rows as case-insensitive
Please answer these questions before submitting your issue. Thanks!

1. What version of Python are you using?

 `Python 3.10.8 (main, Oct 13 2022, 09:48:40) [Clang 14.0.0 (clang-1400.0.29.102)]`

2. What operating system and processor architecture are you using?

`macOS-13.3.1-arm64-arm-64bit`

3. What are the component versions in the environment (`pip freeze`)?

... (Snowpark 1.5.1)

4. What did you do?

```
df.collect(case_sensitive=False)
df.to_local_iterator()  # No such flag
```

5. What did you expect to see?

Support for constructing case-insensitive rows (the underlying function, `self._session._conn.execute` supports the combination of to_iter=True and case_sensitive=False, looks like just an API addition)

6. Can you set logging to DEBUG and collect the logs?

...

### PR DESCRIPTION
**Pull request by cindykrafft**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #926

2. Fill out the following pre-review checklist:

   - [X] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Adds case_sensitive keyword arg and passes through to `self._session._conn.execute`.

### CODE REVIEW

**Comment by cindykrafft** [comment]
I have read the CLA Document and I hereby sign the CLA

File: tests/integ/test_dataframe.py [hunk 1]
```diff
@@ -1956,6 +1956,50 @@ def test_case_insensitive_collect(session):
...
+def test_case_insensitive_local_iterator(session):
+    df = session.create_dataframe(
+        [["Gordon", 153]], schema=["firstname", "matches_won"]
+    )
+    df_quote = session.create_dataframe(
+        [["Gordon", 153]], schema=["'quotedName'", "quoted-won"]
+    )
+
+    # tests for sync collect
+    row = list(df.to_local_iterator(case_sensitive=False))[0]
```
Comment (sfc-gh-stan) [inline tests/integ/test_dataframe.py:1968]: NIT: I prefer using `next` instead of `list(...)[0]`

File: tests/integ/test_dataframe.py [hunk 2]
```diff
@@ -1956,6 +1956,50 @@ def test_case_insensitive_collect(session):
...
+    assert row.MaTchEs_WoN == 153
+    assert row["matches_won"] == 153
+    assert row["Matches_Won"] == 153
+    assert row["MATCHES_WON"] == 153
+
+    with pytest.raises(
+        ValueError,
+        match="Case insensitive fields is not supported in presence of quoted columns",
+    ):
+        row = list(df_quote.to_local_iterator(case_sensitive=False))[0]
```
Comment (sfc-gh-stan) [inline tests/integ/test_dataframe.py:1987]: NIT: we don't need `row = ` here.

File: src/snowflake/snowpark/dataframe.py [hunk 3]
```diff
@@ -696,6 +699,8 @@ def to_local_iterator(
             block: A bool value indicating whether this function will wait until the result is available.
                 When it is ``False``, this function executes the underlying queries of the dataframe
                 asynchronously and returns an :class:`AsyncJob`.
+            case_sensitive: A bool value which is controls the case sensitivity of the fields in the
```
[the remark this answers is not in the document]
Comment (cindykrafft) [reply src/snowflake/snowpark/dataframe.py]: Done

File: tests/integ/test_dataframe.py (hunk 1 above)
Comment (cindykrafft) [reply tests/integ/test_dataframe.py]: Done

File: tests/integ/test_dataframe.py (hunk 2 above)
Comment (cindykrafft) [reply tests/integ/test_dataframe.py]: Done

File: tests/integ/test_dataframe.py [hunk 4]
```diff
@@ -1956,6 +1956,50 @@ def test_case_insensitive_collect(session):
...
+
+    # special character tests
+    df_login = session.create_dataframe(
+        [["admin", "test"], ["snowman", "test"]], schema=["username", "p@$$w0rD"]
+    )
+    row = list(df_login.to_local_iterator(case_sensitive=False))[0]
```
[the remark this answers is not in the document]
Comment (cindykrafft) [reply tests/integ/test_dataframe.py]: Done

**Review by sfc-gh-stan — APPROVED** [review]
The changes LGTM, I've kicked off the merge gates in https://github.com/snowflakedb/snowpark-python/pull/1099

**Comment by sfc-gh-aling** [comment]
thanks for the PR, can you also help update the changelog? https://github.com/snowflakedb/snowpark-python/blob/main/CHANGELOG.md

**Review by sfc-gh-aling — APPROVED** [review]

**Comment by cindykrafft** [comment]
> thanks for the PR, can you also help update the changelog? https://github.com/snowflakedb/snowpark-python/blob/main/CHANGELOG.md

Done

**Review by sfc-gh-sfan — APPROVED** [review]

**Pull request merged**